static/to_compile/src/search_solution_form_controller.ts

```diff
@@ -427,4 +433,29 @@ export default class extends Controller<HTMLElement> {
             this.searchFormTarget.dispatchEvent(submitEvent)
         }, 300)
     }
+
+    toggleAPropos() {
+        if (this.aProposMainPanelTarget.classList.contains("qfdmo-hidden")) {
+            this.#showAPropos()
+        } else {
+            this.#hideAPropos()
+        }
+        this.scrollToContent()
```

Comment:
`setTimeout` appelée dans showAPropos et hideAPropos est asynchrone. 
Ce qui veut dire qu'on va scroller jusqu'au contenu avant la fin de l'animation.  
Cf https://developer.mozilla.org/fr/docs/Web/API/setTimeout 

100ms ça devrait pas être bloquant, mais si on allonge l'animation cela pourrait amener des problèmes, genre scroller au mauvais endroit 

--- 
Solutions possibles : 
- Rendre showAPropos, toggleAPropos asynchrones et attendre l'exécution de `this.$showAPropos` avec `await this.#showAPropos()` (je suis pas 100% sûr de la méthode à utiliser, il semblerait qu'il faille utiliser une promise https://stackoverflow.com/questions/33289726/combination-of-async-function-await-settimeout) 
- Ajouter un paramètre à `showAPropos` et `hideAPropos` qui soit une fonction appelée après le trigger des classes, dans le `setTimeout`, en l'occurence `this.scrollToContent`

Reply:
ok pour moi, mon explication est claire ou bof ?

Reply:
j'en ai compris l'esprit :)